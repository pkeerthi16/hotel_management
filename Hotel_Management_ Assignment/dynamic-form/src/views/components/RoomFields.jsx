import React from 'react';
import PropTypes from 'prop-types';
import { CHILD_ACCOMODATES, ADULT_ACCOMODATES } from '../../constants';
import { RoomContainer, SelectContainer, Header, CustomCheckbox, Field, Select } from './StyledComponents';


const RoomFields = (props) =>
    <RoomContainer>
        <Header className="header">
            {props.firstRoom ? null : <CustomCheckbox name="roomSelected" checked={props.selected} onChange={(event) => props.onCheck(props.roomNo, event.target.checked)} />}
            <h4>Room {props.roomNo}</h4>
        </Header>
        <SelectContainer bgColor={props.selected ? 'white' : 'transparent'}>
            <Field>
                <label>Adults (18+)</label>
                <Select disabled={!props.selected} name="numberOfAdults" value={props.numberOfAdults} onChange={(event) => props.onNumberSelect(props.roomNo, parseInt(event.target.value), 'adult')}>
                    {
                        ADULT_ACCOMODATES.map(val =>
                            <option key={val} value={val}>{val}</option>
                        )
                    }
                </Select>
            </Field>
            <Field>
                <label>Children (0-17)</label>
                <Select disabled={!props.selected} name="numberOfChilds" value={props.numberOfChildren} onChange={(event) => props.onNumberSelect(props.roomNo, parseInt(event.target.value), 'child')}>
                    {
                        CHILD_ACCOMODATES.map(val =>
                            <option key={val} value={val}>{val}</option>
                        )
                    }
                </Select>
            </Field>
        </SelectContainer>
    </RoomContainer>;

RoomFields.propTypes = {
    firstRoom: PropTypes.bool,
    selected: PropTypes.bool.isRequired,
    numberOfAdults: PropTypes.number.isRequired,
    numberOfChildren: PropTypes.number.isRequired,
    onCheck: PropTypes.func,
    onNumberSelect: PropTypes.func,
}

RoomFields.defaultProps = {
    firstRoom: false,
    selected: false,
    numberOfAdults: 1,
    numberOfChildren: 0,
}

export default RoomFields;