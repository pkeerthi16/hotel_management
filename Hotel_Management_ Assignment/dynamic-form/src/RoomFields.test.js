import React from 'react';
import ReactDOM from 'react-dom';
import RoomFields from './views/components/RoomFields';
import renderer from 'react-test-renderer';


it('RoomFields renders without crashing with props', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RoomFields firstRoom={false} numberOfAdults={3} numberOfChildren={2} selected={true} />, div);
  ReactDOM.unmountComponentAtNode(div);
})

it('RoomFields renders without crashing, without props', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RoomFields />, div);
  ReactDOM.unmountComponentAtNode(div);
})

it('RoomFields snapshot test with selected: true', () => {
  const tree = renderer.create(<RoomFields firstRoom={false} numberOfAdults={2} numberOfChildren={2} selected={true} />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('RoomFields snapshot test with selected: false', () => {
  const tree = renderer.create(<RoomFields firstRoom={false} numberOfAdults={2} numberOfChildren={2} selected={false} />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('Verify select inputs are disabled on selected: false', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RoomFields selected={false} />, div);
  const selectElements = div.getElementsByTagName('select');
  expect(selectElements[0].disabled).toBe(true);
  expect(selectElements[1].disabled).toBe(true);
})

it('Verify select inputs are enabled on selected: true', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RoomFields selected={true} />, div);
  const selectElements = div.getElementsByTagName('select');
  expect(selectElements[0].disabled).toBe(false);
  expect(selectElements[1].disabled).toBe(false);
})
