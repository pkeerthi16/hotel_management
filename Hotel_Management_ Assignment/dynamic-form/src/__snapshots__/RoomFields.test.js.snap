// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomFields snapshot test with selected: false 1`] = `
<div
  className="sc-bxivhb gzioTq"
>
  <div
    className="header sc-ifAKCX fnpglF"
  >
    <input
      checked={false}
      className="sc-EHOje fLSRiA"
      name="roomSelected"
      onChange={[Function]}
      type="checkbox"
    />
    <h4>
      Room 
    </h4>
  </div>
  <div
    className="sc-bZQynM bFZbwp"
  >
    <div
      className="sc-gzVnrw bfFLzg"
    >
      <label>
        Adults (18+)
      </label>
      <select
        className="sc-htoDjs fxfdfZ"
        disabled={true}
        name="numberOfAdults"
        onChange={[Function]}
        value={2}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
      </select>
    </div>
    <div
      className="sc-gzVnrw bfFLzg"
    >
      <label>
        Children (0-17)
      </label>
      <select
        className="sc-htoDjs fxfdfZ"
        disabled={true}
        name="numberOfChilds"
        onChange={[Function]}
        value={2}
      >
        <option
          value={0}
        >
          0
        </option>
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`RoomFields snapshot test with selected: true 1`] = `
<div
  className="sc-bxivhb gzioTq"
>
  <div
    className="header sc-ifAKCX fnpglF"
  >
    <input
      checked={true}
      className="sc-EHOje fLSRiA"
      name="roomSelected"
      onChange={[Function]}
      type="checkbox"
    />
    <h4>
      Room 
    </h4>
  </div>
  <div
    className="sc-bZQynM fmpwfT"
  >
    <div
      className="sc-gzVnrw bfFLzg"
    >
      <label>
        Adults (18+)
      </label>
      <select
        className="sc-htoDjs fxfdfZ"
        disabled={false}
        name="numberOfAdults"
        onChange={[Function]}
        value={2}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
      </select>
    </div>
    <div
      className="sc-gzVnrw bfFLzg"
    >
      <label>
        Children (0-17)
      </label>
      <select
        className="sc-htoDjs fxfdfZ"
        disabled={false}
        name="numberOfChilds"
        onChange={[Function]}
        value={2}
      >
        <option
          value={0}
        >
          0
        </option>
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
      </select>
    </div>
  </div>
</div>
`;
